ui/StatusQ/src/statuswindow.cpp

```diff
@@ -1,43 +1,30 @@
 #include "StatusQ/statuswindow.h"
 
 StatusWindow::StatusWindow(QWindow *parent)
-: QQuickWindow(parent),
-  m_isFullScreen(false)
+: QQuickWindow(parent)
 {
     removeTitleBar();
 
     connect(this, &QQuickWindow::windowStateChanged, [&](Qt::WindowState windowState) {
-        if (windowState == Qt::WindowNoState) {
+        if (windowState == Qt::WindowNoState) { // windowed
             removeTitleBar();
-            m_isFullScreen = false;
-            emit isFullScreenChanged();
         } else if (windowState == Qt::WindowFullScreen) {
-            m_isFullScreen = true;
-            emit isFullScreenChanged();
             showTitleBar();
         }
     });
 }
 
-void StatusWindow::updatePosition()
+void StatusWindow::restoreWindowState()
 {
-    auto point = QPoint(screen()->geometry().center().x() - geometry().width() / 2,
-                        screen()->geometry().center().y() - geometry().height() / 2);
-    if (point != this->position()) {
-        this->setPosition(point);
-    }
+    setWindowStates(windowStates() & ~Qt::WindowMinimized);
 }
 
 void StatusWindow::toggleFullScreen()
 {
-    if (m_isFullScreen) {
-        showNormal();
-    } else {
-        showFullScreen();
-    }
+    setWindowStates(windowStates() ^ Qt::WindowFullScreen);
```

Comment:
In the #14132 ticket I had a problem with the `Maximized` state. The FullScreen state usually creates a separate desktop on mac

Reply:
But this is about the "fullscreen" state, which is a separate thing from maximized. Anyways, pls give it a try if it also solves your problem; if not, yell back :)

Reply:
I've tested the branch and it works pefectly!

Reply:
I noticed another glitch with the double title bar  (first I clicked on full screen, then the "exit full screen" button).

<img width="1225" alt="image" src="https://github.com/status-im/status-desktop/assets/1083341/f8fffec9-0255-480a-a979-66b888da5553">

Reply:
OK, that _might_ be something I introduced :/ Can you test on master without this PR? I don't have access to a mac

Reply:
I've tested on master and there is no double title bar when normalising from full screen. 

The system just makes a small jump: 

https://github.com/status-im/status-desktop/assets/1083341/732d40fa-12d5-48fb-a88a-382ce47e16df

Reply:
Thanks for testing! I'll fix the double window decoration

Reply:
@friofry pls update and give it another try, thanks :)

Reply:
the fix works 👍